src/test/java/com/scottlogic/filters/combine/AndCombineFiltersTest.java

```diff
@@ -0,0 +1,128 @@
+
+
+
+    @Test
+    void filter_null_null() {
+        when(localAuthorPostFilter.filter(null)).thenReturn(null);
+        when(localDatePostFilter.filter(null)).thenReturn(null);
+
+        List<UserPost> actual = new AndCombineFilters(localAuthorPostFilter, localDatePostFilter).filter(null);
+
+        assertNull(actual);
+    }
+
+    @Test
+    void filter_EmptyList_EmptyList() {
+        List<UserPost> inputList = Collections.emptyList();
+        List<UserPost> expected = Collections.emptyList();
+
+        when(localAuthorPostFilter.filter(inputList)).thenReturn(List.of());
+
+        when(localDatePostFilter.filter(List.of())).thenReturn(List.of());
+
+        List<UserPost> actual = new AndCombineFilters(localAuthorPostFilter, localDatePostFilter).filter(inputList);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void filter_1ItemListMatchBoth_1ItemList() {
+        List<UserPost> inputList = List.of(userPostJane);
+        List<UserPost> expected = List.of(userPostJane);
+
+        when(localAuthorPostFilter.filter(inputList)).thenReturn(List.of(userPostJane));
+        when(localDatePostFilter.filter(List.of(userPostJane))).thenReturn(List.of(userPostJane));
+
+        List<UserPost> actual = new AndCombineFilters(localAuthorPostFilter, localDatePostFilter).filter(inputList);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void filter_1Item1Match_EmptyList() {
+        List<UserPost> inputList = List.of(userPostJane);
+        List<UserPost> expected = List.of();
+
+        when(localAuthorPostFilter.filter(inputList)).thenReturn(List.of(userPostJane));
+        when(localDatePostFilter.filter(List.of(userPostJane))).thenReturn(List.of());
+
+        List<UserPost> actual = new AndCombineFilters(localAuthorPostFilter, localDatePostFilter).filter(inputList);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void combine_3Items1Match_1ItemList() {
+        List<UserPost> inputList = Arrays.asList(userPostJane, userPostJoe1, userPostJoe2);
+        List<UserPost> expected = List.of(userPostJane);
+
+        when(localAuthorPostFilter.filter(inputList)).thenReturn(Arrays.asList(userPostJane, userPostJoe1));
+        when(localDatePostFilter.filter(Arrays.asList(userPostJane, userPostJoe1))).thenReturn(List.of(userPostJane));
+
+        List<UserPost> actual = new AndCombineFilters(localAuthorPostFilter, localDatePostFilter).filter(inputList);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void combine_3Items2Match_2ItemList() {
+        List<UserPost> inputList = Arrays.asList(userPostJane, userPostJoe1, userPostJoe2);
+        List<UserPost> expected = List.of(userPostJane, userPostJoe2);
+
+        when(localAuthorPostFilter.filter(inputList)).thenReturn(Arrays.asList(userPostJane, userPostJoe1, userPostJoe2));
+        when(localDatePostFilter.filter(Arrays.asList(userPostJane, userPostJoe1, userPostJoe2))).thenReturn(List.of(userPostJane, userPostJoe2));
```

Comment:
One the really long lines like this one sometimes it is worth changing it to be over two lines to increase readability for example:
```
when(localDatePostFilter.filter(Arrays.asList(userPostJane, userPostJoe1, userPostJoe2)))
       .thenReturn(List.of(userPostJane, userPostJoe2));
```
can then improve the readability (though my spacing might be slightly off). This creates a clear delineation between the input and output which makes them both easier to read and keeps the lines shorter which is generally more polite

Reply:
Yeah that would definitely help thanks